tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/ServerlessMiniAgent.cs

```diff
@@ -0,0 +1,164 @@
+// <copyright file="ServerlessMiniAgent.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.ServerlessInstrumentation;
+
+internal static class ServerlessMiniAgent
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ServerlessMiniAgent));
+
+    internal static string GetMiniAgentPath(PlatformID os, ImmutableTracerSettings settings)
+    {
+        if (!settings.IsRunningInGCPFunctions && !settings.IsRunningInAzureFunctionsConsumptionPlan)
+        {
+            return null;
+        }
+
+        string rustBinaryPath;
+        if (Environment.GetEnvironmentVariable("DD_MINI_AGENT_PATH") != null)
+        {
+            rustBinaryPath = Environment.GetEnvironmentVariable("DD_MINI_AGENT_PATH");
+        }
+        else
+        {
+            // Environment.OSVersion.Platform can return PlatformID.Unix on MacOS, this is OK as GCP & Azure don't have MacOs functions.
+            if (Environment.OSVersion.Platform != PlatformID.Unix && Environment.OSVersion.Platform != PlatformID.Win32NT)
+            {
+                Log.Error("Serverless Mini Agent is only supported on Windows and Linux.");
+                return null;
+            }
+
+            var isWindows = os == PlatformID.Win32NT;
+            var dirPathSep = isWindows ? "\\" : "/";
+
+            string rustBinaryPathRoot;
+            if (settings.IsRunningInGCPFunctions)
+            {
+                rustBinaryPathRoot = "/layers/google.dotnet.publish/publish/bin";
+            }
+            else
+            {
+                // IsAzureFunction
+                if (isWindows)
+                {
+                    rustBinaryPathRoot = "C:\\home\\site\\wwwroot";
+                }
+                else
+                {
+                    // linux
+                    rustBinaryPathRoot = "/home/site/wwwroot";
+                }
+            }
+
+            string rustBinaryPathOsFolder = isWindows ? "datadog-serverless-agent-windows-amd64" : "datadog-serverless-agent-linux-amd64";
+            string rustBinaryFileExtension = isWindows ? ".exe" : string.Empty;
+            rustBinaryPath = string.Format("{0}{1}{2}{3}datadog-serverless-trace-mini-agent{4}", rustBinaryPathRoot, dirPathSep, rustBinaryPathOsFolder, dirPathSep, rustBinaryFileExtension);
```

Comment:
We should use `System.IO.Path.Combine()` instead of concatenating by hand.

Reply:
fixed